src/main/kotlin/net/axay/kspigot/localization/KSpigotLocalization.kt

```diff
@@ -0,0 +1,78 @@
+package net.axay.kspigot.localization
+
+import org.apache.commons.lang.text.StrSubstitutor
+import org.bukkit.entity.Player
+import java.io.InputStreamReader
+import java.nio.charset.StandardCharsets
+import java.util.*
+import kotlin.collections.HashMap
+
+/**
+ * Handles localization of strings using java [ResourceBundle]s.
+ *
+ * Message property files should reside in `src/main/resources` named
+ * `messages_locale.properties`, where `locale` is formatted as returned by
+ * [Locale.toString].
+ */
+object L10n {
+    /**
+     * This function determines which locale is used for a player, by default
+     * [Locale.US] is returned.
+     *
+     * It is invoked every time a message is localized, so some sort of caching
+     * is advisable in many cases.
+     */
+    var localeProvider: (Player) -> Locale = { Locale.US }
```

Comment:
In this way, isn't the locale always set to "US"?

Reply:
As stated by the documentation this is the default behavior, the user however free to change it any time, being able to choose a players locale in any desirable way.

Maybe there could be some standard providers to choose from implemented later on.